Utilities/ocvalidate/ValidateKernel.c

```diff
@@ -215,39 +215,39 @@ CheckKernelAdd (
         "Kernel->Add[%u]->BundlePath (length %u) is too long (should not exceed %u)!\n",
         Index,
         AsciiStrLen (BundlePath),
-        OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH)
+        OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH) - 1
         ));
       ++ErrorCount;
     }
 
     //
     // There is one missing '\\' after the concatenation of BundlePath and ExecutablePath. Append one.
     //
-    ExecutableFixedSize   = L_STR_LEN (OPEN_CORE_KEXT_PATH) + AsciiStrLen (BundlePath) + 1;
-    ExecutablePathSumSize = ExecutableFixedSize + AsciiStrSize (ExecutablePath);
+    ExecutableFixedSize   = AsciiStrLen (BundlePath) + 1 + AsciiStrLen (ExecutablePath);
+    ExecutablePathSumSize = L_STR_LEN (OPEN_CORE_KEXT_PATH) + ExecutableFixedSize;
     if (ExecutablePathSumSize > OC_STORAGE_SAFE_PATH_MAX) {
       DEBUG ((
         DEBUG_WARN,
-        "Kernel->Add[%u]->ExecutablePath (length %u) is too long (should not exceed %u)!\n",
+        "Kernel->Add[%u]->BundlePath + ExecutablePath (length %u) is too long (should not exceed %u)!\n",
```

Comment:
Also coming round to your way of thinking. How about printing `BundlePath + '\' + ExecutablePath` in the message (so adding `+ '\\' ` to the format string here, almost what you had before)?